app/services/messages/send_push.rb

```diff
@@ -11,10 +11,8 @@ def self.call(*args)
     end
 
     def call
-      receiver_ids = chat_channel.chat_channel_memberships.
-        where.not(user_id: user.id).pluck(:user_id)
-
-      PushNotificationSubscription.where(user_id: receiver_ids).find_each do |sub|
+      receivers = chat_channel.chat_channel_memberships.where.not(user_id: user.id).select(:user_id)
+      PushNotificationSubscription.where(user_id: receivers).find_each do |sub|
```

Comment:
Thanks to the magic of lazy loading, receivers will not be loaded and instead the query executed will be something like:

```sql
SELECT "push_notification_subscriptions".* FROM "push_notification_subscriptions" 
WHERE "push_notification_subscriptions"."user_id" IN (
 SELECT "chat_channel_memberships"."user_id" FROM "chat_channel_memberships" 
 WHERE "chat_channel_memberships"."chat_channel_id" = 3 
 AND "chat_channel_memberships"."user_id" != 111)
```

effectively using a single query